pkg/envoy/generator/lds/accesslog.go

```diff
@@ -0,0 +1,231 @@
+package lds
+
+import (
+	"encoding/json"
+	"fmt"
+
+	xds_accesslog "github.com/envoyproxy/go-control-plane/envoy/config/accesslog/v3"
+	xds_core "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	xds_grpc_accesslog "github.com/envoyproxy/go-control-plane/envoy/extensions/access_loggers/grpc/v3"
+	xds_otel_accesslog "github.com/envoyproxy/go-control-plane/envoy/extensions/access_loggers/open_telemetry/v3"
+	xds_accesslog_stream "github.com/envoyproxy/go-control-plane/envoy/extensions/access_loggers/stream/v3"
+	otlpCommon "go.opentelemetry.io/proto/otlp/common/v1"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/types/known/anypb"
+	"google.golang.org/protobuf/types/known/structpb"
+
+	"github.com/openservicemesh/osm/pkg/models"
+	"github.com/openservicemesh/osm/pkg/protobuf"
+)
+
+var (
+	defaultAccessLogJSONFields = &structpb.Struct{
+		Fields: map[string]*structpb.Value{
+			"start_time":            structpb.NewStringValue(`%START_TIME%`),
+			"method":                structpb.NewStringValue(`%REQ(:METHOD)%`),
+			"path":                  structpb.NewStringValue(`%REQ(X-ENVOY-ORIGINAL-PATH?:PATH)%`),
+			"protocol":              structpb.NewStringValue(`%PROTOCOL%`),
+			"response_code":         structpb.NewStringValue(`%RESPONSE_CODE%`),
+			"response_code_details": structpb.NewStringValue(`%RESPONSE_CODE_DETAILS%`),
+			"time_to_first_byte":    structpb.NewStringValue(`%RESPONSE_DURATION%`),
+			"upstream_cluster":      structpb.NewStringValue(`%UPSTREAM_CLUSTER%`),
+			"response_flags":        structpb.NewStringValue(`%RESPONSE_FLAGS%`),
+			"bytes_received":        structpb.NewStringValue(`%BYTES_RECEIVED%`),
+			"bytes_sent":            structpb.NewStringValue(`%BYTES_SENT%`),
+			"duration":              structpb.NewStringValue(`%DURATION%`),
+			"upstream_service_time": structpb.NewStringValue(`%RESP(X-ENVOY-UPSTREAM-SERVICE-TIME)%`),
+			"x_forwarded_for":       structpb.NewStringValue(`%REQ(X-FORWARDED-FOR)%`),
+			"user_agent":            structpb.NewStringValue(`%REQ(USER-AGENT)%`),
+			"request_id":            structpb.NewStringValue(`%REQ(X-REQUEST-ID)%`),
+			"requested_server_name": structpb.NewStringValue("%REQUESTED_SERVER_NAME%"),
+			"authority":             structpb.NewStringValue(`%REQ(:AUTHORITY)%`),
+			"upstream_host":         structpb.NewStringValue(`%UPSTREAM_HOST%`),
+		},
+	}
+
+	defaultAccessLogJSONStr = protobuf.MustToJSON(defaultAccessLogJSONFields)
+)
+
+type accessLogBuilder struct {
+	name                    string
+	format                  string
+	openTelemetryCluster    string
+	openTelemetryAttributes map[string]string
+}
+
+// NewAccessLogBuilder returns an accessLogBuilder instance used to build access log configuration
+func NewAccessLogBuilder() *accessLogBuilder { //nolint: revive // unexported-return
```

Comment:
Can we add some tests that given a accessLogBuilder with a set of parameters, tests the `Build` method?